docs/contracts/errors.rst

```diff
@@ -38,9 +40,43 @@ caller.
         // ...
     }
 
+And the same again with the ``require`` statement:
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.26;
+
+    /// Insufficient balance for transfer. Needed `required` but only
+    /// `available` available.
+    /// @param available balance available.
+    /// @param required requested amount to transfer.
+    error InsufficientBalance(uint256 available, uint256 required);
+
+    // This will only compile with via-ir
+    contract TestToken {
+        mapping(address => uint) balance;
+        function transfer(address to, uint256 amount) public {
+            require(amount > balance[msg.sender], InsufficientBalance(balance[msg.sender], amount));
```

Comment:
Yup, I could rename the function in the original example to `transferWithRevertError` and paste this one in the above contract as `transferWithRequireError`, but then I'd have to bump the version to `0.8.26` and slap an `// This will only compile with via-ir`, which would be confusing with regards to the `revert InsufficientBalance{...}` function. What do you think?

Reply:
I think it would be fine. You can always clarify it by adding a short comment above the `require` saying that this is the new form, available only via IR and above `if` that it works both with IR and legacy.

I think that the overall improvement in readability is worth it. As a reader I think I'd rather have it in one snippet where I can clearly see what differs rather than have two separate snippets I have to compare.

Reply:
I dunno, I honestly think that cramming both examples into one and then having to differentiate them with in code comments just to avoid and extra 20 lines help with readability.

Reply:
We often use comments in examples this way, highlighting various details, so this seems perfectly natural to me. On the other hand using two full examples for such a tiny detail is just unnecessary padding. Making it longer does not make it more readable.

Another issue here is that documenting it this way puts emphasis on the wrong thing. The example we need here is one that shows the use of custom errors. This can be easily done in one function using both `require` and `revert` - would actually be best to rewrite it that way. The fact that you can also use the `if ... revert` pattern instead of `require` is just a footnote. It could by itself be just a comment or a note below the example.